src/test/epoll/CMakeLists.txt

```diff
@@ -1,43 +1,9 @@
-## dependency on glib
 include_directories(${GLIB_INCLUDES})
 link_libraries(${GLIB_LIBRARIES})
 
-## create and install an executable that can run outside of shadow
 add_executable(test-epoll test_epoll.c)
-add_executable(test-epoll-writeable test_epoll_writeable.c)
-
-## register the tests
-add_test(NAME epoll COMMAND test-epoll)
-
-
-add_test(
-    NAME epoll-shadow-ptrace
-    COMMAND sh -c "\
-    ${yaml2xml} ${CMAKE_CURRENT_SOURCE_DIR}/epoll.test.shadow.config.yaml --output - \
-    | ${CMAKE_BINARY_DIR}/src/main/shadow --interpose-method=ptrace -l debug -d epoll-ptrace.shadow.data - \
-    "
-)
-add_test(
-    NAME epoll-shadow-preload
-    COMMAND sh -c "\
-    ${yaml2xml} ${CMAKE_CURRENT_SOURCE_DIR}/epoll.test.shadow.config.yaml --output - \
-    | ${CMAKE_BINARY_DIR}/src/main/shadow --interpose-method=preload -l debug -d epoll-preload.shadow.data - \
-    "
-)
-
-
-add_test(
-    NAME epoll-writeable-shadow-ptrace
-    COMMAND sh -c "\
-    ${yaml2xml} ${CMAKE_CURRENT_SOURCE_DIR}/epoll-writeable.test.shadow.config.yaml --output - \
-    | ${CMAKE_BINARY_DIR}/src/main/shadow --interpose-method=ptrace -l debug -d epoll-writeable-ptrace.shadow.data - \
-    "
-)
-add_test(
-    NAME epoll-writeable-shadow-preload
-    COMMAND sh -c "\
-    ${yaml2xml} ${CMAKE_CURRENT_SOURCE_DIR}/epoll-writeable.test.shadow.config.yaml --output - \
-    | ${CMAKE_BINARY_DIR}/src/main/shadow --interpose-method=preload -l debug -d epoll-writeable-preload.shadow.data - \
-    "
-)
+add_linux_tests(BASENAME epoll COMMAND test-epoll)
+add_shadow_tests(BASENAME epoll METHODS ptrace-only ptrace preload)
 
+add_executable(test-epoll-writeable test_epoll_writeable.c)
+add_shadow_tests(BASENAME epoll-writeable METHODS ptrace-only ptrace preload)
```

Comment:
Can you add a TODO note here about running the `test-epoll-writeable` test outside of Shadow? I don't see why it was not previously enabled, and I think it should be using `launch_with_unused_ports.py` and `test-launcher`.